features/steps/backup-restore.py

```diff
@@ -146,14 +173,47 @@ def step_impl(context):
 
 @given(u'I have a private public key pair')
 def step_impl(context):
-    raise NotImplementedError(u'STEP: Given I have a private public key pair')
+
+    c = gpg.Context(armor=True)
+
+    context.gpgdir = TemporaryDirectory()
+    c.home_dir = context.gpgdir.name
+    userid = "backup-" + ''.join([random.choice(string.ascii_letters + string.digits)
+                                  for n in range(10)])
+
+    c.create_key(userid, algorithm="rsa3072", expires_in=31536000, encrypt=True)
+
+    context.public_key = c.key_export_minimal(pattern=userid)
+    context.private_key = c.key_export_secret(pattern=userid)
 
 
 @when(u'I run a backup on the database using the public key')
 def step_impl(context):
-    raise NotImplementedError(u'STEP: When I run a backup on the database using the public key')
+    call_ansible_step("given that I have configured environment definitions",
+                      playbook="test-backup.yml")
+    keystr = context.public_key.decode('utf-8').replace('\n', '\\n')
+    run_fargate_with_envfile('backup-task-environment', image="paddlehq/mysql-backup-s3",
+                             call_env=["-e", "PUBLIC_KEY=" + keystr])
 
 
 @when(u'I restore that backup to a new database using the private key')
 def step_impl(context):
-    raise NotImplementedError(u'STEP: When I restore that backup to a new database using the private key' )
+    call_ansible_step("given that I have configured environment definitions",
+                      playbook="test-backup.yml")
+    keystr = context.private_key.decode('utf-8').replace('\n', '\\n')
+    run_fargate_with_envfile('restore-task-environment', image="paddlehq/s3-restore-mysql",
+                             call_env=["-e", "PRIVATE_KEY=" + keystr])
+
+
+def verify_s3_object_encrypted(object):
+    """needless to say, checking the name of an object is not the best way to see if
```

Comment:
we need to check this in a immediate future. right now, boto3 doesn't have any check for the client side encryption, only for the server side encryption. But this could change fast, since it's one of the few SDK's that doesn't have it. https://docs.aws.amazon.com/general/latest/gr/aws_sdk_cryptography.html